src/plugins/profiling/server/routes/search_flamechart.ts

```diff
@@ -71,63 +66,79 @@ export function registerFlameChartSearchRoute(router: IRouter<DataRequestHandler
     },
     async (context, request, response) => {
       const { index, projectID, timeFrom, timeTo } = request.query;
-      const sampleSize = 200;
+      const targetSampleSize = 20000; // minimum number of samples to get statistically sound results
+      const topN = 200; // collect data for the top N unique stack traces
 
       try {
         const esClient = context.core.elasticsearch.client.asCurrentUser;
         const filter = newProjectTimeQuery(projectID!, timeFrom!, timeTo!);
 
-        // const resp = await getCountResponse(context, filter);
+        // Start with counting the results in the index down-sampled by 5^6.
+        // That is in the middle of our down-sampled indexes.
+        const pow6 = 6;
         const resp = await esClient.search({
-          index: 'profiling-events-1000',
+          index: 'profiling-events-5pow' + pow6,
           body: {
             query: filter,
             size: 0,
             track_total_hits: true,
           },
         });
+        const sampleCountFromPow6 = resp.body.hits.total.value as number;
```

Comment:
I have two concerns about the variables `pow6` and `sampleCountFromPow6`:

* the special constant in the variables may require the variables to be renamed if the number of down-sampled indices changes or the starting position changes in the future
* initial reader confusion if accompanying comments are missed

I think we may be able to address both concerns if we use slightly more verbose names, e.g.

* `initialPow` and `sampleCountFromInitialPow`
* `startingPow` and `sampleCountFromStartingPow`
* `defaultPow` and `sampleCountFromDefaultPow`
* `initialExponent` and `sampleCountFromInitialExponent`
* `startingExponent` and `sampleCountFromStartingExponent`
* `defaultExponent` and `sampleCountFromDefaultExponent`

These examples are only suggestions that first came to mind, so feel free to try other names that are a better fit.

Reply:
Thanks, amended. Also renamed more vars and added consts for readability.